### PR TITLE
backend/feat: #2085 integrate location tracking service

### PR DESCRIPTION
**Pull request by chakri642**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable
- [ ] No leak detected in leakcanary

### CODE REVIEW

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Environment.hs [hunk 1]
```diff
@@ -180,7 +186,12 @@ data AppEnv = AppEnv
     enableAPILatencyLogging :: Bool,
     enableAPIPrometheusMetricLogging :: Bool,
     eventStreamMap :: [EventStreamMap],
-    locationTrackingServiceKey :: Text
+    locationTrackingServiceKey :: Text,
+    ltsCfg :: LocationTrackingeServiceConfig,
+    enableLocationTrackingStartRide :: Bool,
```
Comment (khuzema786) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Environment.hs:191]: Make it into one single key

File: Backend/lib/location-updates/src/Lib/LocationUpdates/Internal.hs [hunk 2]
```diff
@@ -104,7 +127,7 @@ recalcDistanceBatches h@RideInterpolationHandler {..} ending driverId = do
   updateDistance driverId distanceToUpdate
   where
     atLeastBatchPlusOne = getWaypointsNumber driverId <&> (> batchSize)
-    pointsRemaining = (> 0) <$> getWaypointsNumber driverId
+    pointsRemaining = (> 0) <$> getWaypointsNumber driverId -- What do this do ?
```
Comment (khuzema786) [inline Backend/lib/location-updates/src/Lib/LocationUpdates/Internal.hs:130]: Remove this comment

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/SharedLogic/DriverPool.hs [hunk 3]
```diff
@@ -475,13 +485,44 @@ calculateDriverPool poolStage driverPoolCfg mbVariant pickup merchantId onlyNotO
...
+  m [NearestDriversResult]
+getNearestDriversLocs mbVariant LatLong {..} radiusMeters merchantId onlyNotOnRide mbDriverPositionInfoExpiry = do
+  logDebug $ "getNearestDriversLocs xyz" <> show mbDriverPositionInfoExpiry <> show mbVariant <> show LatLong {..} <> show radiusMeters <> show merchantId <> show onlyNotOnRide
+  enableLocationTrackingNearByRide <- asks (.enableLocationTrackingNearByRide)
+  driverLocs <- do
+    if enableLocationTrackingNearByRide
+      then do
+        ltsCfg <- asks (.ltsCfg)
+        a <- LF.nearBy ltsCfg lat lon mbVariant radiusMeters merchantId
+        logDebug $ "driverLocs abc" <> show a
```
Comment (shanm-16) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/SharedLogic/DriverPool.hs:505]: logdebug?

Comment (chakri642) [reply Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/SharedLogic/DriverPool.hs:505]: i will remove

**Review by khuzema786 — APPROVED** [review]

**Pull request closed without merging**